app/views/devise/sessions/phone_code.html.erb

```diff
@@ -13,6 +13,8 @@
     <% end %>
 
     <%= form_with url: user_session_phone_verify_path, method: :post, html: { autocomplete: "off", "data-module" => "explicit-cross-domain-links" } do %>
+      <%= hidden_field_tag "from_confirmation_email", params[:from_confirmation_email] %>
```

Comment:
Lower down this page, at

```
    <p class="govuk-body">
      <%= sanitize(t("mfa.phone.code.not_received.sign_in_message", link: user_session_phone_resend_path)) %>
    </p>
```

We'll lose the param if the user follows the link.

Reply:
Good spot.  I've pushed the parameter through on this link too.